File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution
{
public:
    void moveZeroes(vector<int> &nums)
    {
        vector<int> output;
        output.reserve(nums.size());
        int count = 0;
        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] == 0)
                count++;
            if (nums[i])
                output.push_back(nums[i]);
        }
        for (int i = 0; i < count; i++)
            output.push_back(0);
        nums = output;
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
# time complexity: O(n)
# space complexity: O(1)
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left = 0
        for right in range(len(nums)):
            if nums[right] != 0 and nums[left] == 0:
                nums[right], nums[left] = nums[left], nums[right]
            if nums[left] != 0:
                left += 1
        return nums